tests/test_tag.py

```diff
@@ -52,77 +49,71 @@ def test(self):
+        ret = main(["tag", "list"])
+        self.assertEqual(ret, 0)
 
-            self.assertEqual("", logger.handlers[0].stream.getvalue())
+        self.assertEqual("", self._caplog.text)
 
         ret = main(["tag", "add", "v1"])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_standard_output()
+        self._caplog.clear()
 
-            ret = main(["tag", "list"])
-            self.assertEqual(ret, 0)
+        ret = main(["tag", "list"])
+        self.assertEqual(ret, 0)
 
-            self.assertEqual(
-                logger.handlers[0].stream.getvalue(),
-                "bar.dvc:\n"
-                "  bar:\n"
-                "    v1:\n"
-                "      md5: 8978c98bb5a48c2fb5f2c4c905768afa\n"
-                "foo.dvc:\n"
-                "  foo:\n"
-                "    v1:\n"
-                "      md5: acbd18db4cc2f85cedef654fccc4a4d8\n"
-                "\n",
-            )
+        expected = (
+            "bar.dvc:\n"
+            "  bar:\n"
+            "    v1:\n"
+            "      md5: 8978c98bb5a48c2fb5f2c4c905768afa\n"
+            "foo.dvc:\n"
+            "  foo:\n"
+            "    v1:\n"
+            "      md5: acbd18db4cc2f85cedef654fccc4a4d8\n"
+        )
+
+        # TODO: Format `caplog` output to capture only "%(message)s"
+        stdout = "\n".join(record.message for record in self._caplog.records)
+
+        assert expected == stdout
 
         ret = main(["tag", "remove", "v1"])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_standard_output()
+        self._caplog.clear()
 
-            ret = main(["tag", "list"])
-            self.assertEqual(ret, 0)
+        ret = main(["tag", "list"])
+        self.assertEqual(ret, 0)
 
-            self.assertEqual("", logger.handlers[0].stream.getvalue())
+        self.assertEqual("", self._caplog.text)
 
 
 class TestTagAddNoChecksumInfo(TestDvc):
     def test(self):
         ret = main(["run", "-o", self.FOO, "--no-exec"])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_error_output()
+        self._caplog.clear()
 
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
             ret = main(["tag", "add", "v1", "foo.dvc"])
             self.assertEqual(ret, 0)
 
-            self.assertEqual(
-                "Warning: missing checksum info for 'foo'\n",
-                logger.handlers[1].stream.getvalue(),
-            )
+        assert "missing checksum info for 'foo'" in self._caplog.text
 
 
 class TestTagRemoveNoTag(TestDvc):
     def test(self):
         ret = main(["add", self.FOO])
         self.assertEqual(ret, 0)
 
-        with MockLoggerHandlers(logger), ConsoleFontColorsRemover():
-            reset_logger_error_output()
+        self._caplog.clear()
```

Comment:
Why couldn't `self._caplog.at_level` down below clear log by default?

Reply:
`_caplog.at_level` is only a context manager to set the level of a logger, and changing the level shouldn't affect the previous log, what `caplog.clear()` does is to remove all the `LogRecord` entries saved on `caplog.records`.

Reply:
@mroutis yeah, but why would you use at_level and not use clear() before it? I'm not saying at_level should do that perse, but it feels like it would be convenient to have something that would reset it automatically when you are capturing log with `with`. What do you think?

Reply:
I guess `at_level` is to test logging specific info, for example, it is used in `test_logger.py` for logic that depends on the current level (outputting the traceback on an error only when `--verbose` is specified).

In this test, we were running the `add` command before, creating some unwanted logs for our test:
```python
    def test(self):
        ret = main(["add", self.FOO])
        self.assertEqual(ret, 0)
```

That's why I do:
```python
        self._caplog.clear()
```

Now, the `at_level` thing is because the previous test was asserting that the output should contain `Warning: <message>`, to maintain compatibility, I used `_caplog.at_level` to make sure the message didn't come from _debug_ or _info_:
```
        with self._caplog.at_level(logging.WARNING, logger="dvc"):
            ret = main(["tag", "remove", "v1", "foo.dvc"])
            self.assertEqual(ret, 0)

        assert "tag 'v1' not found for 'foo'" in self._caplog.text
```

I don't think this is always the case, as sometimes we would be trying to test something logged by `info`. Also, I'd prefer to leave it as it is, you can read and write tests without having to know beforehand about a `set_level_and_clear` method, for example :stuck_out_tongue: